bluemira/geometry/parameterisations.py

```diff
@@ -986,6 +986,191 @@ def _make_control_points(p0, p3, theta0, theta3, l_start, l_end):
         return p1, p2
 
 
+class PictureFrameTools:
+    """
+    Curved Pictureframe parameterisation with straight (flat) inner limb
+
+    Parameters
+    ----------
+    var_dict: Optional[dict]
+        Dictionary with which to update the default values of the parameterisation.
+
+    """
+
+    @staticmethod
+    def _make_domed_leg(
+        axis,
+        x_out,
+        x_curve_start,
+        x_mid,
+        z_top,
+        z_mid,
+        flip=False,
+        r_c=0,
+    ):
+        """
+        Makes smooth dome for CP coils
+
+        Returns
+        -------
+        shape: BluemiraWire
+            CAD Wire of the geometry
+
+        """
+        # Define basic Top Curve (with no joint or corner transitions)
+
+        out_wire = []
+
+        # Define the basic main curve (with no joint or transitions curves)
+        alpha = np.arctan(0.5 * (x_out - x_curve_start) / abs(z_top - z_mid))
+        theta_leg_basic = 2 * (np.pi - 2 * alpha)
+        r_leg = 0.5 * (x_out - x_curve_start) / np.sin(theta_leg_basic / 2)
+        z_top_r_leg = z_top + r_leg if flip else z_top - r_leg
+        leg_centre = (x_out - 0.5 * (x_out - x_curve_start), 0, z_top_r_leg)
+
+        # Transitioning Curve
+        sin_a = np.sin(theta_leg_basic / 2)
+        cos_a = np.cos(theta_leg_basic / 2)
+        alpha_leg = (
+            np.arcsin(np.abs(r_leg * sin_a - r_c) / (r_leg - r_c)) + theta_leg_basic / 2
+        )
+
+        # Joint Curve
+        r_j = min(x_curve_start - x_mid, 0.8)
+        theta_j = np.arccos((r_leg * cos_a + r_j) / (r_leg + r_j))
+        z_mid_r_j = z_mid - r_j if flip else z_mid + r_j
+        joint_curve_centre = (
+            leg_centre[0] - (r_leg + r_j) * np.sin(theta_j),
+            0,
+            z_mid_r_j,
+        )
+
+        # Final total angular extent of main curve
+        theta_leg_final = alpha_leg - (theta_leg_basic / 2 - theta_j)
+
+        # Build straight section of leg
+        p1 = [x_mid, 0, z_mid]
+        p2 = [leg_centre[0] - (r_leg + r_j) * np.sin(theta_j), 0, z_mid]
+
+        if not flip:
+            out_wire.append(make_polygon([p1, p2]))
+        else:
+            out_wire.append(make_polygon([p2, p1]))
+        out_wire.append(
```

Comment:
Is is still the straight section?

Seems to be the curve joining the main dome with the straight section (please add a `# Arc joining the straight section with the dome arc`)